app/models/user.js

```diff
@@ -21,5 +24,9 @@ export default Addressable.extend({
 
   fullName: Ember.computed('firstName', 'lastName', function(){
     return (this.get('firstName') + " " + this.get('lastName'));
+  }),
+
+  isInfoComplete: Ember.computed('firstName', 'lastName', 'email', 'mobile', function(){
+    return this.get('firstName') && this.get('firstName').length !== 0 && this.get('lastName') && this.get('lastName').length !== 0 && this.get('email') && this.get('email').length !== 0 && this.get('mobile') && this.get('mobile').length !== 0 && this.get('title') && this.get('title').length !== 0;
```

Comment:
An empty string evaluates to false, so there is no need to check for the length as well.
You can check that by running `Boolean("") => false`

You could replace 
`this.get('firstName') && this.get('firstName').length !== 0 && ...`
by
`this.get('firstName') && ...`

Although checking the length would be a good practice in most languages, so I'm not explicitly opposed to it, it does show caution which is not a bad thing.